nomad/client_alloc_endpoint.go

```diff
@@ -430,6 +431,31 @@ func (a *ClientAllocations) exec(conn io.ReadWriteCloser) {
 		return
 	}
 
+	// Handle job ID if requested.
```

Comment:
If the original HTTP API request is made to the client that has the allocation, it's handled entirely locally without going through the server. This extra validation here gets avoids sending the client RPC if it shouldn't be sent, but maybe we should check the alloc client status as long as we're here too? (Recognizing that the allocation could fail after we send the RPC to the client, of course!)

Reply:
Ah yeah, I forgot to mention that this endpoint is kind of "weird" in that sense: it can be routed from servers or sent directly to clients.

I tried to push as much validation to clients, but since they don't have access to cluster state, checking for [`req.JobID != alloc.JobID`](https://github.com/hashicorp/nomad/pull/18904/files/d53bebe75cf65518aa0b614835463ecdf0d218e9#diff-e2b2a83c57344644a143d5dbe1b6fdb28b05316170410efa904a6875d4484db8R252-R255) was all that I could do 😅 

This means that, when sending a request directly to clients, the job ID is not fully checked. But I think that's OK, since matching the alloc `JobID` carries a lot of preconceived assumptions (like that the job exists).

It also felt to me that the API design is a bit strange since the only things we actually care about to run an action is an alloc ID, a task name, and an action, so to me it feels like that this should be a `/v1/allocation/<id>/action?task=<task>&action=<action>` instead of `/v1/job/<job>/action?alloc=<alloc>&task=<task>&action=<action>` since `<job>` is kind of meaningless, but I didn't want to make any big change like this so late into the game 😅

Reply:
Added check for terminal status in 4e2696cc3d